staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -332,6 +334,10 @@ func dedupDeltas(deltas Deltas) Deltas {
 	a := &deltas[n-1]
 	b := &deltas[n-2]
 	if out := isDup(a, b); out != nil {
+		// `a` and `b` are duplicates. Only keep the one returned from isDup().
+		// TODO: This extra array allocation and copy seems unnecessary if
```

Comment:
The thing proposed by this TODO is potentially risky depending on who might have taken a pointer. I'd omit or include a disclaimer / rephrase to "might be worth profiling and investigating if it is safe to optimize"

Reply:
Added disclaimer. I had no intention to make premature optimization unless profiling reveals this is a bottleneck for memory allocation.